flyteidl/protos/flyteidl/admin/agent.proto

```diff
@@ -91,3 +91,21 @@ message DeleteTaskRequest {
 // Response to delete a task.
 message DeleteTaskResponse {
 }
+
+// A message used to do a task
+message DoTaskRequest {
```

Comment:
Lets merge the DoTaskRequest and DoTaskResponse into CreateTask
Make 
```
message CreateTaskResponse{
  oneof res {
   bytes resource_meta;
   Resource resource;
  }
}
```

After this we should modify the entire 
message Resource to have more fields that map TaskExecutionEvent

Add fileds like
https://github.com/flyteorg/flyte/blob/b14f1ac1c69b5925440ca847f6a85083b64bb854/flyteidl/protos/flyteidl/event/event.proto#L254
```
    // log information for the task execution
    repeated core.TaskLog logs = 6;

    // This timestamp represents when the original event occurred, it is generated
    // by the executor of the task.
    google.protobuf.Timestamp occurred_at = 7;

    // A predefined yet extensible Task type identifier. If the task definition is already registered in flyte admin
    // this type will be identical, but not all task executions necessarily use pre-registered definitions and this
    // type is useful to render the task in the UI, filter task executions, etc.
    string task_type = 14;

    // Phase associated with the event
    core.TaskExecution.Phase phase = 4;

    // id of the process that sent this event, mainly for trace debugging
    string producer_id = 5;

```

we could skip `task_type` and `occured_at` as this is know to propeller

Reply:
Most importantly i would remove the State and move it to Phase